src/main/java/dao/WriterImpl.java

```diff
@@ -0,0 +1,30 @@
+package dao;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+import service.ReportImpl;
+
+public class WriterImpl implements Writer {
+    private static final String FILE_PATH = "src/main/reports/report.csv";
+    private static final Map<String, Integer> reportMap = new ReportImpl().report();
+
+    @Override
+    public void reportWriter() {
+        String header = "fruit, quantity";
+        try {
+            BufferedWriter writer = new BufferedWriter(new FileWriter(FILE_PATH, true));
+            writer.write(header);
+            writer.newLine();
+            for (Map.Entry<String, Integer> entry : reportMap.entrySet()) {
+                writer.write(entry.getKey());
+                writer.write(", " + entry.getValue());
+                writer.newLine();
+            }
+            writer.close();
```

Comment:
use try with resources instead. Now if we get exception in try block write.close() won't be called and we won't close connection to file